library/core/functions.general.php

```diff
@@ -2152,6 +2152,23 @@ function jsonEncodeChecked($value, $options = null) {
     }
 }
 
+if (!function_exists('jsonFilter')) {
+    /**
+     * Prepare data for json_encode.
+     *
+     * @param mixed $value
+     */
+    function jsonFilter(&$value) {
+        if (is_array($value)) {
+            array_walk_recursive($value, __FUNCTION__);
+        } elseif ($value instanceof \DateTimeInterface) {
+            $value = $value->format('r');
+        } elseif ($ip = ipDecode($value)) {
```

Comment:
This looks iffy to me. I'd at least do a === test on the return. Is there any way we can test for a binary string too? At least do an is_string().